envoy/ssl/handshaker.h

```diff
@@ -90,6 +91,11 @@ class HandshakerFactoryContext {
    * The list of supported protocols exposed via ALPN, from ContextConfig.
    */
   virtual absl::string_view alpnProtocols() const PURE;
+
+  /**
+   * @return reference to the server lifecycle notifier
```

Comment:
It might be helfpul to explain here why this is needed, as there are no callers and no tests.

Reply:
added a test